### PR TITLE
fix(platform/bitbucket): ensure `getPrList()` runtime integrity

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- When PR is updated modify/update it's corresponding obj in pr cache

## Context
- Ref: https://github.com/renovatebot/renovate/issues/32040

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository (locally)

### CODE REVIEW

File: lib/modules/platform/bitbucket/index.ts [hunk 1]
```diff
@@ -1065,6 +1070,14 @@ export async function updatePr({
       `/2.0/repositories/${config.repository}/pullrequests/${prNo}/decline`,
     );
   }
+
+  // update pr cache
+  await BitbucketPrCache.addPr(
```
Comment (rarkins) [inline lib/modules/platform/bitbucket/index.ts:1075]: The add function also does updating too?

Comment (RahulGautamSingh) [reply lib/modules/platform/bitbucket/index.ts:1075]: Yes.

https://github.com/renovatebot/renovate/blob/aabf1638d1a39fa675d2eecc7a23076b378c1f28/lib/modules/platform/bitbucket/pr-cache.ts#L77-L90

Comment (rarkins) [reply lib/modules/platform/bitbucket/index.ts:1075]: Please rename it to setPr()

File: lib/modules/platform/bitbucket/index.spec.ts [hunk 2]
```diff
@@ -1668,7 +1668,12 @@ describe('modules/platform/bitbucket/index', () => {
         .get('/2.0/repositories/some/repo/pullrequests/5')
         .reply(200, { reviewers: [reviewer] })
         .put('/2.0/repositories/some/repo/pullrequests/5')
-        .reply(200);
+        .reply(200, { id: 5 })
+        .get(`/2.0/repositories/some/repo/pullrequests`)
+        .query(true)
```
Comment (viceice) [inline lib/modules/platform/bitbucket/index.spec.ts:1673]: what is this for? to ignore query parameters?

**Review by rarkins — DISMISSED** [review]

File: lib/modules/platform/bitbucket/index.spec.ts [hunk 3]
```diff
@@ -1898,6 +1918,112 @@ describe('modules/platform/bitbucket/index', () => {
...
+              title: 'title',
+            },
+          ],
+        })
+        .get('/2.0/repositories/some/repo/pullrequests/5')
+        .reply(200, { reviewers: [reviewer] })
+        .put('/2.0/repositories/some/repo/pullrequests/5')
+        .reply(200, { id: 5, title: 'newTitle' });
+
+      await bitbucket.getPrList(); // cache is now initialized
```
Comment (viceice) [inline lib/modules/platform/bitbucket/index.spec.ts:2014]: you should validate old title to make it prove for accidental changes

File: lib/modules/platform/bitbucket/pr-cache.ts [hunk 4]
```diff
@@ -74,19 +74,19 @@ export class BitbucketPrCache {
     return prCache.getPrs();
   }
 
-  private addPr(pr: Pr): void {
+  private setPr(pr: Pr): void {
     logger.debug(`Adding PR #${pr.number} to the PR cache`);
```
Comment (viceice) [inline lib/modules/platform/bitbucket/pr-cache.ts:78]: needs a change. this renaming should be a separate PR

File: lib/modules/platform/bitbucket/index.spec.ts (hunk 2 above)
Comment (RahulGautamSingh) [reply lib/modules/platform/bitbucket/index.spec.ts:1673]: No.

When cache is initialized for first time, it's internal method `init()` is called which populates the cache with list of PRs. 

In real runs, `getPrList()` initializes/pupulates the cache at the beginning. But in the tests, `updatePr` is called directly hence we need this mock.

Comment (viceice) [reply lib/modules/platform/bitbucket/index.spec.ts:1673]: i mean the `query(true)`

Comment (RahulGautamSingh) [reply lib/modules/platform/bitbucket/index.spec.ts:1673]: Yes, adding `query(true)` matches the request regardless of what query params are passed.

File: lib/modules/platform/bitbucket/pr-cache.ts (hunk 4 above)
Comment (RahulGautamSingh) [reply lib/modules/platform/bitbucket/pr-cache.ts:78]: https://github.com/renovatebot/renovate/pull/33174

**Comment by RahulGautamSingh** [comment]
Blocked by: https://github.com/renovatebot/renovate/pull/33174

**Review by viceice — APPROVED** [review]
